pydantic/annotated_arguments.py

```diff
@@ -39,22 +39,21 @@ def __get_pydantic_core_schema__(
             return core_schema.no_info_before_validator_function(self.func, schema=schema)  # type: ignore
 
 
-@slots_dataclass
+@slots_dataclass(frozen=True)
 class PlainValidator:
     func: core_schema.NoInfoValidatorFunction | core_schema.GeneralValidatorFunction
 
     def __get_pydantic_core_schema__(
         self, source_type: Any, handler: Callable[[Any], core_schema.CoreSchema]
     ) -> core_schema.CoreSchema:
-        schema = handler(source_type)
         info_arg = inspect_validator(self.func, 'plain')
         if info_arg:
-            return core_schema.general_plain_validator_function(self.func, schema=schema)  # type: ignore
+            return core_schema.general_plain_validator_function(self.func)  # type: ignore
```

Comment:
@adriangb is this correct? Both `general_plain_validator_function` and `no_info_plain_validator_function` actually don't have the `schema` keyword.

Reply:
Yes that's correct. Surprised we did not have a test for this.